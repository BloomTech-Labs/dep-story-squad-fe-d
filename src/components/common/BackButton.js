import React from 'react';
import { Button } from 'antd';
import { ArrowLeftOutlined } from '@ant-design/icons';

const BackButton = props => {
  return (
    <Button className="back-button" type="default" icon={<ArrowLeftOutlined />}>
      Go Back
    </Button>
  );
};
export default BackButton;
