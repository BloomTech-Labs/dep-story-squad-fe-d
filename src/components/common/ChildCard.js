import React from 'react';
import { Card, Button, Layout } from 'antd';

const ChildCard = props => {
  return (
    <Card className="child-card">
      <Layout className="child-info">
        <img src={props.AvatarURL} alt="avatar" />
        <h2
          style={{
            fontFamily: 'Nunito, sans-serif',
            color: 'black',
            fontSize: '1.5rem',
          }}
        >
          {props.name}
        </h2>
      </Layout>
      <Button type="primary" size="large" htmlType="submit">
        {props.name.toUpperCase()}'S PROGRESS
      </Button>
    </Card>
  );
};

export default ChildCard;
