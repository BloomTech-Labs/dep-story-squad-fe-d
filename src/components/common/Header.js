import React from 'react';

import { Link } from 'react-router-dom';
import { Button, Dropdown, Menu } from 'antd';
import { MenuOutlined } from '@ant-design/icons';

const ChildMenu = ({ authService, ...props }) => {
  return (
    <Menu {...props}>
      <Menu.Item key="1">
        <Link to="/child/dashboard">Home</Link>
      </Menu.Item>
      <Menu.Item key="2">
        <Link>Help</Link>
      </Menu.Item>
      <Menu.Item key="3">
        <Link>Change User</Link>
      </Menu.Item>
      <Menu.Item key="4" onClick={() => authService.logout()}>
        Log Out
      </Menu.Item>
    </Menu>
  );
};

const Header = ({ authService = null, displayMenu = true }) => {
  return (
    <div className="hero">
      {displayMenu && (
        <Dropdown
          overlay={<ChildMenu authService={authService} />}
          trigger={['click']}
          className="menu-button"
        >
          <Button icon={<MenuOutlined />} type="text" />
        </Dropdown>
      )}
      <h1 className="header-text">STORY SQUAD</h1>
    </div>
  );
};
export default Header;
