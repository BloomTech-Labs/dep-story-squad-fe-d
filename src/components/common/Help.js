import React from 'react';

const Help = () => {
   
    return (
        <>
      <h1>Contact US</h1>
      <h3>Email us: email@storysquad.com</h3>
      </>
    );
  };
  
  export default Help;