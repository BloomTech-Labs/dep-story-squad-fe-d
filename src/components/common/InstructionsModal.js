import React, { useState, useEffect } from 'react';
import { Modal, Button } from 'antd';

const InstructionsModal = props => {
  const [visible, setVisible] = useState(false);

  const showModal = () => {
    setVisible(true);
  };

  const handleCancel = e => {
    setVisible(false);
  };

  const handleOk = e => {
    console.log(e);
    setVisible(false);
  };
  let showButton = null;
  useEffect(() => {
    showModal();
    if (props.showOkButton) {
      showButton = (
        <Button onClick={handleCancel} className="accept-button">
          I Accept!!!
        </Button>
      );
    }
  }, []);

  return (
    <>
      <Modal
        className="instructions-modal"
        visible={visible}
        keyboard={true}
        width={'70%'}
        onCancel={handleCancel}
        zIndex={2000}
        onOk={handleOk}
        cancelButtonProps={{ disabled: true }}
        footer={null}
        closeIcon={null}
      >
        <p style={props.style}>{props.instructions}</p>
        {showButton}
      </Modal>
    </>
  );
};

export default InstructionsModal;
