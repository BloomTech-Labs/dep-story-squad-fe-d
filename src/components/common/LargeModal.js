import React, { useState, useEffect } from 'react';
import { Modal, Button } from 'antd';

const LargeModal = props => {
  const [visible, setVisible] = useState(false);

  return (
    <>
      <Modal
        title="Hello There"
        // visible={visible}
        width={'90%'}
        // onCancel={handleCancel}
        zIndex={2000}
      >
        Hello
      </Modal>
    </>
  );
};

export default LargeModal;
