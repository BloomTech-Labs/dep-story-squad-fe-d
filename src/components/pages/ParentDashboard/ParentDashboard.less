@import '~antd/dist/antd.less';
@import '~antd/lib/style/themes/default.less';

.container {
  background-color: white;
  margin: 0;
}
.trigger {
  font-size: 18px;
  line-height: 64px;
  cursor: pointer;
  transition: color 0.3s;
}
.ant-menu-item-selected {
  color: #6ceae6;
  border-top: 1px solid black;
  border-bottom: 1px solid black;
}

.trigger a:hover {
  color: #1890ff;
}
.logo {
  height: 30vh;
  border: 2px 2px bold;
}
.welcome {
  padding: 2%;
  margin-top: 10%;
  text-align: center;
  size: 2rem;
}
.menu {
  margin-top: 5%;
}
.children {
  display: flex;
  width: 100%;
  justify-content: flex-start;
  padding: 0 2%;
  background-color: #fff;
  .ant-card {
    display: flex;
    flex-flow: column nowrap;
    align-items: center;
    position: relative;
    background-color: @primary-color;
    height: 30vh;
    margin: 0 2.5vw 2.5vh 0;
    padding: 16px;
    width: 45%;
    border-radius: 14px;
    justify-content: center;
    text-align: center;

    h2 {
      align-self: center;
    }

    .ant-card-body,
    .ant-layout {
      background: none;
    }
    img {
      size: 40%;
      height: 22vh;
      padding: 10%;
    }
    button {
      border-radius: 15px;
      box-sizing: border-box;
      width: 85%;
      padding: 3 3vw;
      height: 5vh;
      margin-bottom: '0';
      text-align: center;
      font-size: 1rem;
      a:hover {
        color: black;
        font: bold;
        border-bottom: 1px bold black;
      }
    }
    .child-info {
      display: flex;
      flex-flow: row nowrap;
      flex-grow: 1;
    }
  }
}
.homescreen-content {
  display: flex;
  justify-content: left;
  flex-direction: row;

  margin: 5%;
}

button a:hover {
  color: #1890ff;
}

.site-layout-background {
  background: #fff;
}
.content {
  background-color: #fff;
}

.title {
  text-align: center;
  font-family: 'bangers';
  color: #0267c1;
  background: none;
}

.ant-layout-sider {
  background: #fff;
  color: #fff;
  line-height: 520px;
  height: 100vh;
  border-right: 2px solid gray;
}
.ant-layout-header {
  margin-top: 2%;
}

[class^='ant-'],
[class*='ant-'],
[class*='ant-'] *::after {
  -webkit-box-sizing: border-box;
  box-sizing: border-box;
  background-color: #fff;
}
