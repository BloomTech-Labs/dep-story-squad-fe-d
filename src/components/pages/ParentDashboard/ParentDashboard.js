import React from 'react';
import { Layout, Menu, Typography, Card } from 'antd';
import { Link } from 'react-router-dom';
import { useOktaAuth } from '@okta/okta-react';

import { PlusCircleFilled } from '@ant-design/icons';
import ChildCard from '../../common/ChildCard';

const { Sider } = Layout;
const { Title } = Typography;

const initialChildren = [
  {
    Name: 'Jackie',
    GradeLevelID: '7',
    PIN: '1234',
    AvatarURL: 'https://picsum.photos/200/300.jpg',
    IsDyslexic: false,
  },
  {
    Name: 'Ryan',
    GradeLevelID: '8',
    PIN: '1212',
    AvatarURL: 'https://picsum.photos/200/300.jpg',
    IsDyslexic: false,
  },
];

const ParentDashboard = props => {
  const { authService } = useOktaAuth();

  const { parent } = props;

  return (
    <>
      <Layout className="parent-dashboard">
        <Sider className="sider" theme="light">
          <div className="logo">
            <Title className="welcome" level={4}>
              Welcome Back
            </Title>
          </div>
          <Menu
            className="menu"
            mode="inline"
            defaultSelectedKeys={['dashboard']}
          >
            <Menu.Item key="dashboard">Dashboard</Menu.Item>
            <Menu.Item key="settings">
              <Link to="/parent/settings">Parent Settings</Link>
            </Menu.Item>
            <Menu.Item key="help">
              <Link to="/parent/help">Help</Link>
            </Menu.Item>
            <Menu.Item onClick={() => authService.logout()} key="logout">
              Log out
            </Menu.Item>
          </Menu>
        </Sider>

        <Layout>
          <Title className="title" style={{ color: '#0267C1' }} level={1}>
            STORY SQUAD
          </Title>
          <div className="children-container">
            <Layout className="children" style={{ flexFlow: 'row wrap' }}>
              {initialChildren.map(child => (
                <ChildCard name={child.Name} AvatarURL={child.AvatarURL} />
              ))}
              <Card>
                <h2>
                  <Link to="/parent/add-child">
                    <PlusCircleFilled /> Add a Child
                  </Link>
                </h2>
              </Card>
            </Layout>
          </div>
        </Layout>
      </Layout>
    </>
  );
};

export default ParentDashboard;
