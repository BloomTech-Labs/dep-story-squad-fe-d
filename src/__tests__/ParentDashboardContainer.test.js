import React from 'react';
import { BrowserRouter as Router } from 'react-router-dom';
import { render, cleanup, waitFor, act } from '@testing-library/react';
import configureStore from 'redux-mock-store';
import { Provider } from 'react-redux';

import { ParentLoadingComponent } from '../components/common';
import ParentDashboardContainer from '../components/pages/ParentDashboard/ParentDashboardContainer';
import { ParentDashboard } from '../components/pages/ParentDashboard';

const mockStore = configureStore([]);
const store = mockStore();

afterEach(cleanup);

jest.mock('@okta/okta-react', () => ({
  useOktaAuth: () => {
    return {
      authState: {
        isAuthenticated: true,
      },
      authService: {
        getUser: () => Promise.resolve({ name: 'someone' }),
      },
    };
  },
}));

// works but has warnings for state and un-mounted components - 100%
describe('<ParentDashboardContainer /> test suite', () => {
  test('container renders without crashing', async () => {
    const { getByText, findByText, queryByText } = render(
      <Router>
        <Provider store={store}>
          {/* <ParentDashboardContainer LoadingComponent={ParentLoadingComponent} /> */}
          <ParentDashboard
            LoadingComponent={ParentLoadingComponent}
            parent={{ children: [] }}
          />
        </Provider>
      </Router>
    );
    let loader = getByText(/loading/i);
    expect(loader).toBeInTheDocument();

    // await waitFor(async () => {
    //   await findByText(/welcome/i);
    // });
    // loader = queryByText(/loading/i);
    // expect(loader).toBeNull();
    await act(() => Promise.resolve());
  });
});

// works but has 0% coverage
// describe('<ParentDashboardContainer /> test suite', () => {
//   test('container renders without crashing', async () => {
//     const {getByText, queryByText} = await render(
//       // <Component parent={{children: []}} />
//       <Provider store = {store}>
//         <ParentDashboard parent={{ children: [] }} store={store} />
//       </Provider>, {wrapper: MemoryRouter}
//     );
//     const welcome = getByText(/welcome/i);
//     expect(welcome.innerHTML).toBe('Welcome Back');
//   });
// });

// works but doesn't need the act at the botto,, nor the query for null
// describe('<ParentDashboardContainer /> test suite', () => {
//   test('container renders without crashing', async () => {
//     const {getByText, queryByText} = await render(
//       // <Component parent={{children: []}} />
//       <Provider store = {store}>
//         <ParentDashboard parent={{ children: [] }} store={store} />
//       </Provider>, {wrapper: MemoryRouter}
//     );
//     const welcome = getByText(/welcome/i);
//     expect(welcome.innerHTML).toBe('Welcome Back');

//         const loader = queryByText(/loading/i);
//     expect(loader).toBeNull();
//     await act(() => Promise.resolve());
//   });
// });

// Original
// import * as React from 'react';
// import { configure, shallow } from 'enzyme';
// import Adapter from 'enzyme-adapter-react-16';
// import { BrowserRouter as Router } from 'react-router-dom';

// import configureStore from 'redux-mock-store';
// import { Provider } from 'react-redux';

// import LoadingComponent from '../components/common/ParentLoadingComponent';
// import RenderParentDashboard from '../components/pages/ParentDashboard/RenderParentDashboard';
// import ParentDashboardContainer from '../components/pages/ParentDashboard/ParentDashboardContainer';
// jest.mock('@okta/okta-react', () => ({
//   useOktaAuth: () => {
//     return {
//       authState: {
//         isAuthenticated: true,
//       },
//       authService: {},
//     };
//   },
// }));

// configure({ adapter: new Adapter() });

// describe('<ParentDashboardContainer />', () => {
//   configure({ adapter: new Adapter() });
//   const mockStore = configureStore([]);
//   const store = mockStore();

//   describe('Render <ParentDashboardContainer />', () => {
//     let shallowWrapper;
//     beforeEach(() => {
//       shallowWrapper = shallow(
//         <Router>
//           <Provider store={store}>
//             <ParentDashboardContainer />
//           </Provider>
//         </Router>
//       ).dive();
//     });

//     it('Find ParentDashboard', () => {
//       expect(shallowWrapper.find(RenderParentDashboard));
//     });
//     it('Find Loading Component', () => {
//       expect(shallowWrapper.find(LoadingComponent));
//     });
//     it('Find ParentDashboardContainer', () => {
//       expect(shallowWrapper).toMatchSnapshot();
//     });
//   });
// });
