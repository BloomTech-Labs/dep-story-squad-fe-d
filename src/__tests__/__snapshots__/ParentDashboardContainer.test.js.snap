// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ParentDashboardContainer /> Render <ParentDashboardContainer /> Find ParentDashboardContainer 1`] = `ShallowWrapper {}`;
